src/api/objects/ObjectAPI.js

```diff
@@ -189,6 +189,16 @@ define([
         });
     };
 
+    ObjectAPI.prototype.search = function* (options) {
+        for (const provider of Object.values(this.providers)) {
+            if (provider.search !== undefined) {
+                yield provider.search(options);
```

Comment:
So I think our search providers will also need to be able to yield in order to support search backends that either stream search results or return them in pages (we may hit a page limit in YAMCS for example). 

I'm a generator function n00b, but it looks as though generator functions can return generator functions. I haven't dug too deeply into the practical implications of this though. Basically we want the `.search` function to be able to yield arrays of objects from search providers before they've finished searching, if they support it.

If we can make this optional (and not force generator functions on plugins unless they want to use them) then great, otherwise no big deal if it's mandatory.

Reply:
Also, now that I've thought about this a little more, this is going to block until the first provider yields something before it even queries subsequent providers right? That's not what we want.

Is there an approach that will allow us yield results from any of the search providers as soon as they're ready? I'm way into unknown territory here...Maybe generator functions aren't the right way of handling this, maybe we should just use event emitters?

Reply:
As we discussed, the generator yields promises so the yield values are ready immediately. The unit tests show object provider searches are run individually and non-blocking.

I do have a slight concern and might have to go back at some point to address handling network errors and such.